google/cloud/aiplatform/models.py

```diff
@@ -15,9 +15,213 @@
 # limitations under the License.
 #
 
+from google.cloud.aiplatform_v1beta1.services.model_service import (
+    client as model_client,
+)
 
-class Model:
-    pass
+from typing import Dict, Optional, Sequence
+
+from google.auth import credentials as auth_credentials
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform import utils
+from google.cloud.aiplatform_v1beta1.services.model_service.client import (
+    ModelServiceClient,
+)
+from google.cloud.aiplatform_v1beta1.types import model as gca_model
+from google.cloud.aiplatform_v1beta1.types import env_var
+
+
+class Model(base.AiPlatformResourceNoun):
+
+    client_class = ModelServiceClient
+    is_prediction_client = False
+
+    @property
+    def uri(self):
+        """Uri of the model."""
+        return self._gca_resource.artifact_uri
+
+    @property
+    def description(self):
```

Comment:
Is this the general pattern we will follow, where every top level attribute of the protobuf message object gets promoted to a python object property?  if so we might do this just once in the resource base class (and carefully add correct docstrings).

Reply:
It's a general pattern. I didn't add `description` to the ABC because it's not shared by all Nouns. For example, TrainingPipeline.